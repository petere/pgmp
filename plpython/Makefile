MODULE_big = pgmp_plpython$(python_majorversion)
OBJS = pgmp_plpython.o

PG_CPPFLAGS = -I$(srcdir)/../src $(python_includespec)

EXTENSION = pgmp_plpythonu
DATA = pgmp_plpythonu--1.0.sql

REGRESS = pgmp_plpython

PG_CONFIG = pg_config
PGXS := $(shell $(PG_CONFIG) --pgxs)
include $(PGXS)
