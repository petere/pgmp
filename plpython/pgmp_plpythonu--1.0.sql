CREATE FUNCTION mpz_to_plpython(val mpz) RETURNS internal
LANGUAGE C STRICT IMMUTABLE
AS 'MODULE_PATHNAME';

CREATE TRANSFORM FOR mpz LANGUAGE plpythonu (
    FROM SQL WITH FUNCTION mpz_to_plpython(mpz)
);
