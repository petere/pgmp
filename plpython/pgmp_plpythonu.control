comment = 'transforms between pgmp and plpythonu'
default_version = '1.0'
module_pathname = '$libdir/pgmp_plpython2'
relocatable = true
requires = 'pgmp,plpythonu'
