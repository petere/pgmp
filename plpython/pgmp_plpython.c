#include "postgres.h"
#include "fmgr.h"
#include "plpython.h"
#include "pmpz.h"

PG_MODULE_MAGIC;

PG_FUNCTION_INFO_V1(mpz_to_plpython);
Datum mpz_to_plpython(PG_FUNCTION_ARGS);

Datum
mpz_to_plpython(PG_FUNCTION_ARGS)
{
	const mpz_t z;
    char	   *buf;
	PyObject   *po;

	/* see pmpz_out() */
	PGMP_GETARG_MPZ(z, 0);
	buf = palloc(mpz_sizeinbase(z, 10) + 2);
	po = PyInt_FromString(mpz_get_str(buf, 10, z), NULL, 10);

	return PointerGetDatum(po);
}
