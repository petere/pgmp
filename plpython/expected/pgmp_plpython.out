CREATE EXTENSION plpythonu;
CREATE EXTENSION pgmp;
CREATE EXTENSION pgmp_plpythonu;
CREATE FUNCTION py_add(a mpz, b mpz) RETURNS mpz
LANGUAGE plpythonu
AS $$
return a + b
$$;
SELECT py_add('123'::mpz, '456'::mpz);
 py_add 
--------
 579
(1 row)

